cli/src/cmd/forge/verify.rs

```diff
@@ -177,6 +228,41 @@ impl VerifyArgs {
         Ok(())
     }
 
+    async fn verify_with_sourcify(self) -> eyre::Result<()> {
+        let url = format!(
+            "https://sourcify.dev/server/check-by-addresses?addresses={}&chainIds={}",
+            hex::encode(self.address.as_bytes()),
+            self.chain.id(),
+        );
+        let response = get(url).await?;
+        if !response.status().is_success() {
+            eprintln!(
+                "Sourcify verification request for address ({}) failed with status code {}",
+                hex::encode(self.address.as_bytes()),
+                response.status()
+            );
+            std::process::exit(1);
+        };
+
+        // Response looks like:
+        // `[{"address":"0x68B1D87F95878fE05B998F19b66F4baba5De1aed","status":"false"}]`
+        let parsed_response = response.json::<serde_json::Value>().await?;
```

Comment:
can we add proper bindings for this?
https://app.quicktype.io/ generates rust types for json